stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/endpoint/DetachServiceTest.java

```diff
@@ -0,0 +1,94 @@
+
+        //Invoke recently attached mock2 services. Test detaching and re attaching mock3 service
+        cMsg = MessageUtils.generateHTTPMessage("/mock2", HttpConstants.HTTP_METHOD_GET);
+        response = Services.invoke(SERVICE_EP_PORT, cMsg);
+        Assert.assertNotNull(response, "Response message not found");
+        Assert.assertEquals((int) response.getHttpStatusCode(), 200);
+        Assert.assertEquals(ResponseReader.getReturnValue(response), "Mock2 resource was invoked");
+
+        //Invoke recently attached mock3 service. That detached the mock2 service and attach mock3
+        cMsg = MessageUtils.generateHTTPMessage("/mock3", HttpConstants.HTTP_METHOD_GET);
+        response = Services.invoke(SERVICE_EP_PORT, cMsg);
+        Assert.assertNotNull(response, "Response message not found");
+        Assert.assertEquals((int) response.getHttpStatusCode(), 200);
+        Assert.assertEquals(ResponseReader.getReturnValue(response), "Mock3 invoked. Mock2 detached. Mock4 attached");
+
+        //Invoke detached mock2 services expecting a 404
+        cMsg = MessageUtils.generateHTTPMessage("/mock2", HttpConstants.HTTP_METHOD_GET);
+        response = Services.invoke(SERVICE_EP_PORT, cMsg);
+        Assert.assertNotNull(response, "Response message not found");
+        Assert.assertEquals((int) response.getHttpStatusCode(), 404);
+        Assert.assertEquals(ResponseReader.getReturnValue(response), "no matching service found for path : /mock2");
+
+        //Invoke mock3 services again expecting a error for re-attaching already available service
+        cMsg = MessageUtils.generateHTTPMessage("/mock3", HttpConstants.HTTP_METHOD_GET);
+        response = Services.invoke(SERVICE_EP_PORT, cMsg);
+        Assert.assertNotNull(response, "Response message not found");
+        Assert.assertEquals((int) response.getHttpStatusCode(), 500);
+        Assert.assertEquals(ResponseReader.getReturnValue(response),
+                            "service registration failed: two services have the same basePath : '/mock4'");
+
+        //Invoke mock4 service. mock2 service is re attached
+        cMsg = MessageUtils.generateHTTPMessage("/mock4", HttpConstants.HTTP_METHOD_GET);
+        response = Services.invoke(SERVICE_EP_PORT, cMsg);
+        Assert.assertNotNull(response, "Response message not found");
+        Assert.assertEquals((int) response.getHttpStatusCode(), 200);
+        Assert.assertEquals(ResponseReader.getReturnValue(response), "Mock4 invoked. Mock2 attached");
+
+        //Invoke recently re-attached mock2 services
+        cMsg = MessageUtils.generateHTTPMessage("/mock2", HttpConstants.HTTP_METHOD_GET);
+        response = Services.invoke(SERVICE_EP_PORT, cMsg);
+        Assert.assertNotNull(response, "Response message not found");
+        Assert.assertEquals((int) response.getHttpStatusCode(), 200);
+        Assert.assertEquals(ResponseReader.getReturnValue(response), "Mock2 resource was invoked");
```

Comment:
We can remove duplicate by using a separate method and passing only the changing values as parameters